engine/crates/engine-v2/src/engine.rs

```diff
@@ -221,13 +227,27 @@ impl<R: Runtime> Engine<R> {
                         cache_status: None,
                         client: request_context.client.clone(),
                     },
-                    start.elapsed(),
+                    elapsed,
                 );
             }
-            span.record_gql_status(status);
+
+            if status.is_success() {
+                tracing::Span::current().record_gql_status(status, elapsed, None);
+                tracing::debug!(target: GRAFBASE_TARGET, "router response")
+            } else {
+                let message = response
+                    .first_error_message()
+                    .map(|s| s.into_owned())
+                    .unwrap_or_else(|| String::from("router error"));
+
+                tracing::Span::current().record_gql_status(status, elapsed, Some(message.clone()));
```

Comment:
For tracing, I don't find it obvious that we want the first error message and only that right now. I would not touch it as of now. So `message` should only be used for logs